include/kernel/thread.h

```diff
@@ -280,20 +280,31 @@ static inline bool_t checkBudget(void)
     return false;
 }
 
-/* Everything checkBudget does, but also set the thread
- * state to ThreadState_Restart. To be called from kernel entries
- * where the operation should be restarted once the current thread
- * has budget again.
- */
 
-static inline bool_t checkBudgetRestart(void)
+/* This is called from kernel entries where operation are done only if the
+ * thread still has budget. Otherwise it needs to be restarted first.
+ */
+static inline bool_t checkBudgetWithRestart(void)
 {
+    /* The assumption here is, that an operation was started by the current
+     * thread, thus this thread must be runnable. Otherwise something is wrong
+     * or the assumption does not hold and the function was called in a wrong
+     * context. It's not fatal if this assert is removed in release builds,
+     * because the current thread's state is check before it is set to
+     * ThreadState_Restart.
+     */
     assert(isRunnable(NODE_STATE(ksCurThread)));
-    bool_t result = checkBudget();
```

Comment:
this trivial rewrite will impact the proofs which creates far more work than it's worth.

Reply:
Ok, I was not aware that the proof also care about the local variable names. This was intended to clarify what this state variable actually holds.